src/common.c

```diff
@@ -1100,6 +1100,18 @@ int processors = 1;
 long get_system_cpus(void) {
     processors = 1;
 
+    #ifdef __APPLE__
+        int32_t tmp_processors;
+
+        if (unlikely(GETSYSCTL("hw.logicalcpu", tmp_processors))) {
```

Comment:
So `hw.ncpu` is not what we want to use. I think `hw.logicalcpu ` wins in favor of `hw.physicalcpu ` because that reflects whats in `proc/stat`
I am not sture if `hw.logicalcpu_max` is what we want here because `hw.logicalcpu` may vary depending on the current power management mode. What do you think?